lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -225,16 +225,19 @@ private static function basename(string $filename): string
 
     public static function enableTrackingDeprecations(): void
     {
+        self::$type  = self::$type ?? 0;
```

Comment:
this is needed purely to make psalm happy, because their level 1 is stricter than PHP for what the `|=` operator accept as operand.